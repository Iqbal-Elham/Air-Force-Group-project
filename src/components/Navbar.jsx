import React from 'react';
import { NavLink, Outlet } from 'react-router-dom';
import styles from './styles/Navbar.module.css';
import planet from './styles/images/planet.png';

const Navbar = () => {
  const clickedStyle = ({ isActive }) => ({
    color: isActive ? 'black' : '#a19e9e',
    textDecoration: isActive ? 'underline' : 'none',

  });

  return (
    <>
      <nav className={styles.navbar}>
        <div className={styles.navBrand}>
          <img src={planet} alt="planet logo" width="70" />
          <h1>Space Travelers&apos; Hub</h1>
        </div>
        <ul>
          <li>
            <NavLink style={clickedStyle} to="/">Rockets</NavLink>
          </li>
          <li>
            <NavLink style={clickedStyle} to="/missions">Missions</NavLink>
          </li>
          <li>
            <NavLink style={clickedStyle} to="/my-profile">My Profile</NavLink>
          </li>
        </ul>
      </nav>
      <Outlet />
    </>
  );
};

export default Navbar;
