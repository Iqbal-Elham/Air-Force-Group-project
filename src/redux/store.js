import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
});

export default store;
