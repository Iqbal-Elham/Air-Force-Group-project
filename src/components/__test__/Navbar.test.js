import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../Navbar';
import store from '../../redux/store';

describe('Test the Rocket file', () => {
  it('should match to snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should return the number navlinks', () => {
    render(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    );
    const navlinks = screen.getAllByRole('link');
    expect(navlinks).toHaveLength(3);
  });
});
