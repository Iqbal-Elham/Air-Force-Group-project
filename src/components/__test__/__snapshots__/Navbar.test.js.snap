// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Rocket file should match to snapshot 1`] = `
<nav
  className="navbar"
>
  <div
    className="navBrand"
  >
    <img
      alt="planet logo"
      src="planet.png"
      width="70"
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "underline",
          }
        }
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "color": "#a19e9e",
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className=""
        href="/my-profile"
        onClick={[Function]}
        style={
          Object {
            "color": "#a19e9e",
            "textDecoration": "none",
          }
        }
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
